write-fonts/src/tables/gvar.rs

```diff
@@ -197,19 +198,63 @@ impl GlyphVariations {
         }
     }
 
+    /// Determine if we should use 'shared point numbers'
+    ///
+    /// If multiple tuple variations for a given glyph use the same point numbers,
+    /// it is possible to store this in the glyph table, avoiding duplicating
+    /// data.
+    ///
+    /// This implementation is currently based on the one in fonttools, where it
+    /// is part of the compileTupleVariationStore method:
+    /// <https://github.com/fonttools/fonttools/blob/0a3360e52727cdefce2e9b28286b074faf99033c/Lib/fontTools/ttLib/tables/TupleVariation.py#L641>
+    ///
+    /// # Note
+    ///
+    /// There is likely room for some optimization here, depending on the
+    /// structure of the point numbers. If it is common for point numbers to only
+    /// vary by an item or two, it may be worth picking a set of shared points
+    /// that is a subset of multiple different tuples; this would mean you could
+    /// make some tuples include deltas that they might otherwise omit, but let
+    /// them omit their explicit point numbers.
+    ///
+    /// For fonts with a large number of variations, this could produce reasonable
+    /// savings, at the cost of a significantly more complicated algorithm.
+    fn compute_shared_points(&self) -> Option<PackedPointNumbers> {
+        let mut point_number_counts = HashMap::new();
+        // count how often each set of numbers occurs
+        for deltas in &self.variations {
+            // for each set points, get compiled size + number of occurances
+            let (_, count) = point_number_counts
+                .entry(&deltas.best_point_packing)
+                .or_insert_with(|| {
+                    let size = deltas.best_point_packing.compute_size();
+                    (size as usize, 0usize)
+                });
+            *count += 1;
+        }
+
+        // find the one that saves the most bytes
+        let (pts, (_, count)) = point_number_counts
+            .into_iter()
+            .filter(|(_, (_, count))| *count > 1)
+            .max_by_key(|(_, (size, count))| *count * *size)?;
+
+        // no use sharing points if they only occur once
+        (count > 1).then(|| pts.to_owned())
```

Comment:
Maybe I'm misreading the sea of brackets and underscroes but it seems like you filter for count > 1 and return None otherwise so this has to be true, the comment should move to the filter and this should just say pts.to_owned()?

Reply:
you're right, good catch